devlib/utils/android.py

```diff
@@ -422,23 +422,23 @@ def _ping(device):
 def adb_shell(device, command, timeout=None, check_exit_code=False,
               as_root=False, adb_server=None):  # NOQA
     _check_env()
-    if as_root:
-        command = 'echo {} | su'.format(quote(command))
-    device_part = []
-    if adb_server:
-        device_part = ['-H', adb_server]
-    device_part += ['-s', device] if device else []
-
+    parts = ['adb']
+    if adb_server is not None:
+        parts += ['-H', adb_server]
+    if device is not None:
+        parts += ['-s', device]
+    parts += ['shell',
+              command if not as_root else f'echo {quote(command)} | su']
```

Comment:
I was unable to find documentation stating which (sub-)version of Python 3 devlib was supporting (I might have missed a document or webpage?) so I assumed Py3.6.

You're right that devlib is still supporting Py3.5 but, now that the projects (incl. WA) are slowly moving away from Python 2, it is necessary to define a "minimum" version of Python 3 to be supported. I though Py3.6 made sense as it's fairly common, relatively stable ([2.5 years old; bug-fix support already ended at the end of last year](https://www.python.org/downloads/release/python-360/)) and provides some valuable new features.

The f-strings have been removed from this PR but it is still necessary to define which version of Python 3 is/will be used by the projects; @setrofim & @marcbonnici ?

Reply:
So for this release we will continue to be targeting older version of 3 as well as 2.7. However as @pietos01 mentioned for both WA and devlib, python 2.7 is now deprecated and support will be dropped in the next release. At the point we are enforcing a switch to python 3 we were also thinking of setting Python 3.6 as our new required version as this appears to have a large adoption and introduces a number of worthwhile changes.
I realise this has not been communicated properly so we will need to formalise this information and  get feedback on our plans. @douglas-raillard-arm Do you foresee any issues with this?

Reply:
No worries, it eventually reaches involved people :)

So, in the current LISA landscape:
* Travis' latest GNU/Linux environment is Ubuntu 16,04 xenial which ships with 3.5. Getting higher versions may be possible via some PPA. Travis can also install multiple version of python if we tell it it's a python project, but we can't use that since it also comes with its own (non-optional) virtualenv which conflicts with the one from LISA. (we could configure LISA to cooperate, but we want to check end-to-end install as well in a way that matches what users will do).
* Some LAVA containers/Jenkins environment we use is using Ubuntu Xenial as well. That may be updated, but I would need to have a chat with the infra guys before committing on that. @ionela-voinescu any comment on that ?
* Some people in our team may be using Ubuntu Xenial, I assume this can be extended to other teams, since it will be supported until 2021/2024. 

All in all, we might be able to require Python >=3.6 but that means asking people to use PPA. Do you have some kind of informal list of things that are desirable in 3.6 and not in 3.5 ? (formatted literals are definitely desirable :-)  but may not be big enough to warrant dropping native xenial support :-( ).

Reply:
Ok thank you very much for the information, that's definitely something we'll need to take into account then. 
We don't have such a list yet so we will need to evaluate this. In order to collate this information I've opened the issue https://github.com/ARM-software/devlib/issues/389 to discuss this further.